src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs

```diff
@@ -87,6 +87,37 @@ public MvcOptions()
         /// </summary>
         public FormatterMappings FormatterMappings { get; }
 
+        /// <summary>
+        /// Gets or sets a value which determines how the model binding system interprets exceptions thrown by an <see cref="IInputFormatter"/>.
+        /// The default value of the property is <see cref="InputFormatterExceptionModelStatePolicy.AllExceptions"/>.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This property is associated with a compatibility switch and can provide a different behavior depending on 
+        /// the configured compatibility version for the application. See <see cref="CompatibilityVersion"/> for 
+        /// guidance and examples of setting the application's compatibility version.
+        /// </para>
+        /// <para>
+        /// Configuring the desired of the value compatibility switch by calling this property's setter will take precedence
+        /// over the value implied by the application's <see cref="CompatibilityVersion"/>.
+        /// </para>
+        /// <para>
+        /// If the application's compatibility version is set to <see cref="CompatibilityVersion.Version_2_0"/> then
+        /// this setting will have the value <see cref="InputFormatterExceptionModelStatePolicy.AllExceptions"/> if 
+        /// not explicitly configured.
+        /// </para>
+        /// <para>
+        /// If the application's compatibility version is set to <see cref="CompatibilityVersion.Version_2_1"/> or
+        /// higher then this setting will have the value
+        /// <see cref="InputFormatterExceptionModelStatePolicy.MalformedInputExceptions"/> if not explicitly configured.
+        /// </para>
+        /// </remarks>
+        public InputFormatterExceptionModelStatePolicy InputFormatterExceptionModelStatePolicy
```

Comment:
@Eilon @kichalla - I'm still not happy with this name. I want to change it to `InputFormatterExceptionPolicy` - thoughts?

Reply:
Yes I think that's both reasonable, and a good improvement. I just hope we won't have many more other kinds of input-formatter-exception related policies. That's a risk I'm willing to take. I assume you'll rename the `enum` as well?

Reply:
Yeah, that's the plan